File: frontend/src/types/ticker.ts
export interface Ticker {
  symbol: string;
  name: string;
  sector?: string;
}

export interface TickerSuggestion {
  ticker: string;
  quantity: number;
  action: "HOLD" | "SELL";
  confidence: "high" | "medium" | "low";
  short_term_pct: number | null;
  short_term_label: string | null;
  long_term_pct: number | null;
  long_term_label: string | null;
  price_outlook: string;
  reasoning: string;
}

export interface PortfolioSuggestionsResponse {
  suggestions: TickerSuggestion[];
  market_context: string | null;
}

File: backend/app/schemas/ticker.py
from pydantic import BaseModel


class TickerResponse(BaseModel):
    symbol: str
    name: str
    sector: str | None = None


class TickerSuggestion(BaseModel):
    ticker: str
    quantity: float
    action: str  # "HOLD" | "SELL"
    confidence: str  # "high" | "medium" | "low"
    price_outlook: str
    reasoning: str


class PortfolioSuggestionsResponse(BaseModel):
    suggestions: list["TickerSuggestion"]
    market_context: str | None = None
